lib/segment/src/segment.rs

```diff
@@ -161,7 +161,7 @@ impl SegmentEntry for Segment {
         }
 
         let metric = mertic_object(&self.segment_config.distance);
-        let processed_vector = metric.preprocess(vector.clone());
+        let processed_vector = metric.preprocess(vector).unwrap_or_else(|| vector.to_owned());
```

Comment:
The caller ``preprocess`` needs ``Vec<VectorElementType>`` by value here, but it also became optimized: there is second copying happening anymore in case of ``CosineMetric``